middlewares/validators/taskRequests.ts

```diff
@@ -0,0 +1,76 @@
+import joi from "joi";
+import { TaskRequestResponse, TaskRequestRequest } from "../../types/taskRequests";
+import { NextFunction } from "express";
+import { REQUEST_TYPE, REQUEST_STATE } from "../../constants/requests";
+import { GITHUB_URL } from "../../constants/urls";
+
+import config from "config";
+import { TASK_REQUEST_TYPE } from "../../constants/taskRequests";
+const githubOrg = config.get("githubApi.org");
+const githubBaseUrl = config.get("githubApi.baseUrl");
+const githubIssuerUrlPattern = new RegExp(`^${githubBaseUrl}/repos/${githubOrg}/.+/issues/\\d+$`);
+const githubIssueHtmlUrlPattern = new RegExp(`^${GITHUB_URL}/${githubOrg}/.+/issues/\\d+$`); // Example: https://github.com/Real-Dev-Squad/website-status/issues/1050
+
+export const createTaskRequestValidator = async (
+    req: TaskRequestRequest,
+    res: TaskRequestResponse,
+    next: NextFunction
+) => {
+    const schema = joi
+        .object()
+        .strict()
+        .keys({
+            requestType: joi.string().valid(TASK_REQUEST_TYPE.CREATION, TASK_REQUEST_TYPE.ASSIGNMENT).required().messages({
+                "string.empty": "requestType cannot be empty",
+                "any.required": "requestType is required",
+            }),
+            externalIssueUrl: joi.string().required().regex(githubIssuerUrlPattern).required().messages({
+                "string.empty": "externalIssueUrl cannot be empty",
+                "any.required": "externalIssueUrl is required",
+            }),
+            externalIssueHtmlUrl: joi.string().required().regex(githubIssueHtmlUrlPattern).messages({
+                "string.empty": "externalIssueHtmlUrl cannot be empty",
+                "any.required": "externalIssueHtmlUrl is required",
+            }),
```

Comment:
definitions for `externalIssueUrl` and `externalIssueHtmlUrl` are defined directly within the object schema. These definitions are similar and can be extracted into reusable constants to avoid redundancy or duplication.



My suggestion is to extract the schema definitions for `externalIssueUrl` and `externalIssueHtmlUrl` into constants and use these constants within the main schema definition.




this will improve the maintainability of the code by making it easier to update the schema definitions in one place if needed.

Reply:
I still do not understand can you me an example?

Reply:
In the validator schema, we implement it as defined in the file. If there are any new methods for implementation, please provide documentation link or examples.

Reply:
joi schema definitions for `externalIssueUrl` and `externalIssueHtmlUrl` are repeated. To avoid duplication, you can define these schemas as constants and reuse them in your main schema. This will make the code more maintainable. Here's an example:
 
const externalIssueUrlSchema = joi.string().required().regex(githubIssuerUrlPattern).messages({
  "string.empty": "externalIssueUrl cannot be empty",
  "any.required": "externalIssueUrl is required",
});

const externalIssueHtmlUrlSchema = joi.string().required().regex(githubIssueHtmlUrlPattern).messages({
  "string.empty": "externalIssueHtmlUrl cannot be empty",
  "any.required": "externalIssueHtmlUrl is required",
});

const schema = joi.object().strict().keys({
  requestType: joi.string().valid(TASK_REQUEST_TYPE.CREATION, TASK_REQUEST_TYPE.ASSIGNMENT).required().messages({
    "string.empty": "requestType cannot be empty",
    "any.required": "requestType is required",
  }),
  externalIssueUrl: externalIssueUrlSchema,
  externalIssueHtmlUrl: externalIssueHtmlUrlSchema,
  // other fields...
});

Reply:
You have two pieces of code that are almost the same. It's like if you had to write your name twice on the same page. Instead, we can write it once and use it wherever we need it. Here's how:
```suggestion
   const externalIssueUrlSchema = joi.string().required().regex(githubIssuerUrlPattern).messages({
  "string.empty": "externalIssueUrl cannot be empty",
  "any.required": "externalIssueUrl is required",
});

const externalIssueHtmlUrlSchema = joi.string().required().regex(githubIssueHtmlUrlPattern).messages({
  "string.empty": "externalIssueHtmlUrl cannot be empty",
  "any.required": "externalIssueHtmlUrl is required",
});

// And then use them like this:
externalIssueUrl: externalIssueUrlSchema,
externalIssueHtmlUrl: externalIssueHtmlUrlSchema,

```